test/container-probe.c

```diff
@@ -1,7 +1,63 @@
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+
+void
+error(const char *msg)
+{
+  perror (msg);
+  exit (1);
+}
+
+void
+listen_on_port (int port)
+{
+  int sock_fd, new_sock_fd, portno;
+  socklen_t cli_len;
+  char buffer[256];
+  struct sockaddr_in serv_addr, cli_addr;
+  int n;
+  sock_fd = socket (AF_INET, SOCK_STREAM, 0);
+  if (sock_fd < 0)
+    error("ERROR while opening socket");
+  bzero ((char *) &serv_addr, sizeof (serv_addr));
+  serv_addr.sin_family = AF_INET;
+  serv_addr.sin_addr.s_addr = INADDR_ANY;
+  serv_addr.sin_port = htons (port);
+  if (bind (sock_fd, (struct sockaddr *) &serv_addr, sizeof (serv_addr)) < 0)
+    error("ERROR on binding");
+  listen (sock_fd,5);
+  cli_len = sizeof (cli_addr);
+  new_sock_fd = accept (sock_fd, (struct sockaddr *) &cli_addr, &cli_len);
+  if (new_sock_fd < 0)
+    error("ERROR on accept");
+  bzero (buffer,256);
+  n = write (new_sock_fd,"Sending messages",16);
```

Comment:
Unfortunately this line will probably fail on some CI system somewhere. Unfortunately you need to check for partial writes and EAGAIN. I can link you some code on IRC if this is unfamiliar.